02-peer-protocol.md

```diff
@@ -566,14 +566,17 @@ still be under the maximum message size.  It also ensures that
 a single penalty transaction can spend the entire commitment transaction,
 as calculated in [BOLT #5](05-onchain.md#penalty-transaction-weight-calculation).
 
-### Removing an HTLC: `update_fulfill_htlc` and `update_fail_htlc`
+### Removing an HTLC: `update_fulfill_htlc`, `update_fail_htlc` and `update_fail_malformed_htlc`
 
 For simplicity, a node can only remove HTLCs added by the other node.
 There are three reasons for removing an HTLC: it has timed out, it has
 failed to route, or the payment preimage is supplied.
 
 The `reason` field is an opaque encrypted blob for the benefit of the
-original HTLC initiator as defined in [BOLT #4](04-onion-routing.md).
+original HTLC initiator as defined in [BOLT #4](04-onion-routing.md),
+but there's a special malformed failure variant for the case where
+our peer couldn't parse it; in this case the current node encrypts
```

Comment:
Wait, didn't we discuss this in Milan? I thought we agreed just to send back the hash of the message as we got it from our predecessor, I don't see the advantage in returning the whole packet.